swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/AWS.kt

```diff
@@ -40,3 +87,326 @@ interface AWS {
+    // AWS api doesnt support pagination, breaking down the list because this call can be very expensive
+    imageFilters.forEach {
+      log.info("Getting Images with filter {}", it)
+      val request = DescribeImagesRequest().withFilters(it)
+      val result: DescribeImagesResult = account.ec2(params.region).describeImages(request)
+      images.addAll(convert(result.images))
+    }
+
+    return images.toList()
+  }
+
+  override fun getImage(params: Parameters): AmazonImage? {
+    val account: SpinnakerAccount = findAccount(params) ?: return null
+    val request = DescribeImagesRequest().withImageIds(params.id)
+    val result: DescribeImagesResult = account.ec2(params.region).describeImages(request)
+    return convert<AmazonImage>(result.images).first()
+  }
+
+  override fun getSecurityGroups(params: Parameters): List<AmazonSecurityGroup> {
+    val account: SpinnakerAccount = findAccount(params) ?: return emptyList()
+    val groups = mutableListOf<AmazonSecurityGroup>()
+    var result: DescribeSecurityGroupsResult = account.ec2(params.region)
+      .describeSecurityGroups()
+
+    var nextToken = result.nextToken
+    while (!nextToken.isNullOrBlank()) {
+      result = account.ec2(params.region)
+        .describeSecurityGroups()
+        .withNextToken(nextToken)
+
+      groups.addAll(
+        convert(result.securityGroups)
+      )
+
+      nextToken = result.nextToken
+    }
+
+    return groups
+  }
+
+  override fun getSecurityGroup(params: Parameters): AmazonSecurityGroup? {
+    val account: SpinnakerAccount = findAccount(params) ?: return null
+    val request = DescribeSecurityGroupsRequest().withGroupIds(params.id)
+    val result: DescribeSecurityGroupsResult = account.ec2(params.region).describeSecurityGroups(request)
+    return convert<AmazonSecurityGroup>(result.securityGroups).first()
+  }
+
+  override fun getSnapshots(params: Parameters): List<AmazonSnapshot> {
+    val account: SpinnakerAccount = findAccount(params) ?: return emptyList()
+    val snapshots = mutableListOf<AmazonSnapshot>()
+    val request = DescribeSnapshotsRequest().withMaxResults(snapshotsMaxResult)
+    var result: DescribeSnapshotsResult = account.ec2(params.region)
+      .describeSnapshots(request)
+
+    var nextToken = result.nextToken
+    while (!nextToken.isNullOrBlank()) {
+      request.nextToken = nextToken
+      result = account.ec2(params.region)
+        .describeSnapshots(request)
+
+      snapshots.addAll(
+        convert(result.snapshots)
+      )
+
+      nextToken = result.nextToken
+    }
+
+    return snapshots
+  }
+
+  override fun getSnapshot(params: Parameters): AmazonSnapshot? {
+    val account: SpinnakerAccount = findAccount(params) ?: return null
+    val request = DescribeSnapshotsRequest().withSnapshotIds(params.id)
+    val result = account.ec2(params.region).describeSnapshots(request)
+    return convert<AmazonSnapshot>(result.snapshots).first()
+  }
+
+  private val snapshotsMaxResult = 1000
```

Comment:
should these values come from a config ?

Reply:
same as the other constants. Will address in another changeset